# Cálculo - Tarefa 02

1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  
2- Calcule essa área de forma analítica.

In [ ]:
import numpy as np

In [ ]:
#1)
f = lambda x: x**3 # Função a ser integrada

def calcula_area(a, b, frame_num, func = f):
    num_bins = frame_num + 1 # Quantidade de retângulos
    delta = (b - a) / num_bins # largura do retângulo
    x_cols = np.linspace(a, b - delta, num_bins) + delta / 2 # Divisão do eixo X
    y_cols = f(x_cols) # alturas dos retângulos
    area = np.sum(y_cols * delta) # soma das áreas dos retângulos
    return area
calcula_area(1, 2,1000)

3.7499996257488766

**Resposta 2**  
f(x) = x³, temos: ∫x³ dx = (x⁴)/4 + C

$\int_{a}^{b} f(x) dx = F(b) - F(a)$



Então, a integral definida é: [(2⁴)/4] - [(1⁴)/4] = (16/4) - (1/4) = 15/4 = 3.75.